pdata/pcommon/common.go

```diff
@@ -717,18 +717,14 @@ func (m Map) PutEmptySlice(k string) Slice {
 }
 
 // Sort sorts the entries in the Map so two instances can be compared.
-// Returns the same instance to allow nicer code like:
-//
-//	assert.EqualValues(t, expected.Sort(), actual.Sort())
 //
 // IMPORTANT: Sort mutates the data, if you call this function in a consumer,
 // the consumer must be configured that it mutates data.
-func (m Map) Sort() Map {
+func (m Map) Sort() {
```

Comment:
This should follow the normal deprecation guideline no?

Reply:
Have you read the issue?

Reply:
> Given that most of the usages are done in test code, believe we can make an exception here and introduce a breaking change just by changing the function signature to not return any value.

I did, though the comment ^^ excludes any usage outside of the repos we know about.

I understand this is important to fix and don't want to block that, however it's unclear to me what differentiates this from other breaking changes that require a deprecation.

Reply:
There's still 2 releases this year :)

Reply:
> There's still 2 releases this year :)

That's what I said, 1 month :))

Reply:
Also, a "proper" deprecation is a 3 release step, since we want to get back to have Sort as name (SortWithoutReturn is added and Sort is deprecated; SortWithoutReturn is deprecated and Sort becomes no return; SortWithoutReturn is removed)

Reply:
I think that making it the "right" way significantly complicates the process for downstream distributions. Most of the usages currently don't utilize the returned value (and none of ...Slice.Sort usages do that). So if we go with this change, It'll be mostly no-op or changes like https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16575. If we go with the 2-step deprecation approach, downstream distributions will need to replace _all_ the usages two times: Sort -> Sort<Something> -> Sort. I'm not against that, I just wanted to make sure we all agree on introducing this extra work to the API users and that it's worth avoiding the exception.

Reply:
I am all in favor of the "short" path for this. But looks like @codeboten it is not, so unless he agrees we should do the "right" way.